Refactor tests to use enzyme

aonomike opened the pull request:
- Refactor all tests to use enzyme
Fixes #98 
Co-authored-by: dhemmingson <danhemm@gmail.com>
Co-authored-by: javpet <javorkai.peter@gmail.com>

aonomike committed:
Refactor tests to use enzyme

- Refactor all tests to use enzyme

Co-authored-by: dhemmingson <danhemm@gmail.com>
Co-authored-by: javpet <javorkai.peter@gmail.com>

FedericoEsparza reviewed (changes requested):
@aonomike @dhemmingson @javpet Amazing work! Thanks so much for putting in the work. 

It's a big PR replacing just about the whole test suite, so I have some thoughts and some requests.

Standby...

aonomike commented:
@FedericoEsparza this one was left specifically for you to review :)

aonomike commented:
> @aonomike @dhemmingson @javpet Amazing work! Thanks so much for putting in the work.
> 
> It's a big PR replacing just about the whole test suite, so I have some thoughts and some requests.
> 
> Standby...

Yes the PR is huge and we knew it was gonna be a challenge reviewing. We just decided to mob type during a vs code live share. Maybe next time we break the PRs down?

dhemmingson committed:
Add test for link in artistperformance

aonomike commented:
maybe something that we've noticed is to refactor the `wrapped` to `wrapper` so as to keep up with the conventions

FedericoEsparza commented:
@aonomike This was part of my feedback that I wanted to add. I don't like `wrapped` nor `wrapper`, because it's not immediately clear, sometimes, what you are testing. I prefer using the name of the component, but very importantly LOWERCASED, otherwise it will throw hard to detect errors.

FedericoEsparza commented on src/components/About/About.test.js:
```diff
@@ -1,9 +1,17 @@
+
+  it("displays 'AboutContainer' text", () => {
+    expect(wrapped.find('div').text()).toEqual('AboutContainer')
+  })
```
The beforeEach hook is great forethought, but within the context of this PR, it prematurely might force you to remove the current test (unless you do something like create nested describe blocks).

**I believe the whole current test suite is removed within this PR, which is not good practice, because then replacing it with a simple test, like the following:**
 
```
expect(1+1).toEqual(2)
```

or something else meaningless, like that, could pass the CI and get merged in, if no one is reviewing the PRs properly.

FedericoEsparza commented on src/components/About/About.test.js:
```diff
@@ -1,9 +1,17 @@
-  it ("creates component without crashing", () => {
-      const about = <About/>;
-    });
+describe("<AboutContainer />", () => {
```
Please re-add this test

aonomike commented:
> @aonomike This was part of my feedback that I wanted to add. I don't like `wrapped` nor `wrapper`, because it's not immediately clear, sometimes, what you are testing. I prefer using the name of the component, but very importantly LOWERCASED, otherwise it will throw hard to detect errors.

Maybe to be the devil's advocate, is wrapper not used as a convention in the enzyme community?

FedericoEsparza commented on src/components/Artist/ArtistPerformance/ArtistPerformance.test.js:
```diff
@@ -1,10 +1,26 @@
 
-//Need at least 1 test to pass Travis CI
-
+let wrapped
```
Please move this to just above the `beforeEach` that it corresponds to, and rename it `artistPerformance`.

FedericoEsparza commented on src/components/Artist/ArtistPerformance/ArtistPerformance.test.js:
```diff
@@ -1,10 +1,26 @@
+let wrapped
 describe("<ArtistPerformance />", () => {
-  it ("creates component without crashing", () => {
-    const artistPerformance = <ArtistPerformance />;
```
Please re-add this test.

aonomike committed:
Refactor wrapped to wrapper in test files

aonomike committed:
Merge branch 'develop' of github.com:AgileVentures/sfn-client into 98_refactor-tests-to-use-enzyme

FedericoEsparza commented on src/components/Artist/ArtistPerformance/ArtistPerformance.test.js:
```diff
@@ -1,10 +1,26 @@
-    const artistPerformance = <ArtistPerformance />;
+  beforeEach(()=>{
+    wrapped = shallow( <ArtistPerformance/> )
+  })
```
Please modify `wrapper` to `artistPerformance`

You could add a nested describe block, but that will complicate things for now.

Just be sure to add the `let` right above and outside the `beforeEach`. This won't conflict with the `const` of the same name, because that is within it's own it function scope.

FedericoEsparza commented on src/components/Artist/ArtistPerformance/ArtistPerformance.test.js:
```diff
@@ -1,10 +1,26 @@
+    expect(wrapped.find('Link').children().text()).toContain('More info')
   });
-});
+});
```
Please modify `wrapper` to `artistPerformance`

FedericoEsparza commented on src/components/Artist/ArtistPerformanceList/ArtistPerfomanceList.test.js:
```diff
@@ -1,8 +1,16 @@
 describe('ArtistPerfomanceList', ()=>{
-    it('creates a component without crushing',() => {
-        const artistPerfomanceList = <ArtistPerfomenceList/>
+    beforeEach(()=>{
```
Please re-add test

FedericoEsparza commented on src/components/Artist/ArtistPerformanceList/ArtistPerfomanceList.test.js:
```diff
@@ -1,8 +1,16 @@
+import ArtistPerformance from '../ArtistPerformance/ArtistPerformance'
+import {shallow} from 'enzyme'
+
+let wrapped
```
Please remove let

FedericoEsparza commented on src/components/About/About.test.js:
```diff
@@ -1,9 +1,17 @@
+describe("<AboutContainer />", () => {
+
+  beforeEach(() => {
+    wrapped = shallow( < About / > )
```
Please remove beforeEach.

Please modify to `const about = shallow( <About/> )` and move down to just above the expectation on line 15.
You'll need to change `wrapped` to `about`.

FedericoEsparza commented on src/components/Artist/ArtistPerformanceList/ArtistPerfomanceList.test.js:
```diff
@@ -1,8 +1,16 @@
+    it('has 3 ArtistPerfomence',() => {
+         expect(wrapped.find('ArtistPerformance').length).toEqual(3)
     })
 })
```
Please remove beforeEach.

Please modify to `const artistPerformanceList = shallow( < ArtistPerformanceList/> )` and move down to just above the expectation on line 16.
You'll need to change `wrapper` to `artistPerformanceList`.

FedericoEsparza commented on src/components/Artist/MoneyRaised/MoneyRaised.test.js:
```diff
@@ -1,9 +1,21 @@
 describe("<MoneyRaised />", () => {
-  it("creates component without crashing", () => {
-    const moneyraised = <MoneyRaised />;
-  });
```
Please re-add test, and kindly rename const to `moneyRaised`

FedericoEsparza commented on src/components/Artist/MoneyRaised/MoneyRaised.test.js:
```diff
@@ -1,9 +1,21 @@
+import Chart from "./Chart"
+import {shallow} from "enzyme"
+
+let wrapped;
```
Please move this to just above the `beforeEach` that it corresponds to, and rename it `moneyRaised `.

FedericoEsparza commented on src/components/Artist/MoneyRaised/MoneyRaised.test.js:
```diff
@@ -1,9 +1,21 @@
-  });
+  beforeEach( ()=> {
+    wrapped = shallow(<MoneyRaised/>)
+  })
```
Please modify `wrapper` to `moneyRaised `

FedericoEsparza commented on src/components/Artist/MoneyRaised/MoneyRaised.test.js:
```diff
@@ -1,9 +1,21 @@
+  it("contains the Chart component", ()=> {
+   expect(wrapped.find("Chart").length).toEqual(1) 
+  })
 });
```
Please modify `wrapped` to `moneyRaised `

FedericoEsparza commented on src/components/Artist/SupportedCauses/SupportedCauses.test.js:
```diff
@@ -1,8 +1,13 @@
 import SupportedCauses from './SupportedCauses'
+import {shallow} from 'enzyme'
 
+let wrapped
```
Please remove let

FedericoEsparza commented on src/components/Artist/SupportedCauses/SupportedCauses.test.js:
```diff
@@ -1,8 +1,13 @@
 describe ("<SupportedCauses />", () => {
-  it ("creates a component without crashing", () => {
-      const supportedCause = <SupportedCauses/>;
-  });
```
Please re-add test

FedericoEsparza commented on src/components/Artist/SupportedCauses/SupportedCauses.test.js:
```diff
@@ -1,8 +1,13 @@
+  it('contains text CauseContainer', ()=>{
+    expect(wrapped.find('h3').text()).toEqual('Supported Causes')
+  })
 });
```
Please remove beforeEach.

Please modify this to `const supportedCauses = shallow( < SupportedCauses/> )` and move down to just above the expectation on line 13.
You'll need to change `wrapper` to `supportedCauses `.

FedericoEsparza commented on src/components/Artists/Artists.test.js:
```diff
@@ -1,9 +1,16 @@
 import Artists from "./ArtistsContainer";
+import { shallow } from "enzyme"
+
+let wrapped;
```
Please remove let

FedericoEsparza commented on src/components/Artists/Artists.test.js:
```diff
@@ -1,9 +1,16 @@
 //Need at least 1 test to pass Travis CI
 describe ("<ArtistsContainer />", () => {
-  it ("creates component without crashing", () => {
-    const artists = <Artists/>;
```
Please re-add test

FedericoEsparza commented on src/components/Artists/Artists.test.js:
```diff
@@ -1,9 +1,16 @@
+  it ("contains the text 'ArtistsContainer'", () => {
+    expect(wrapped.find('div').text()).toEqual('ArtistsContainer');
   });
 });
```
Please remove beforeEach.
Modify to `const artists = shallow( < Artists/> )` and move down to just above the expectation on line 14.
You'll need to change `wrapped` to `artists `.

FedericoEsparza commented on src/components/Cause/Cause.test.js:
```diff
@@ -1,9 +1,14 @@
 import Cause from "./CauseContainer";
+import {shallow} from 'enzyme'
 
+let wrapped
```
Please remove let

FedericoEsparza commented on src/components/Cause/Cause.test.js:
```diff
@@ -1,9 +1,14 @@
+  })
+  it('contains text CauseContainer', ()=>{
+    expect(wrapped.text()).toEqual('CauseContainer')
+  })
```
Please remove beforeEach.

Modify to `const cause = shallow( < Cause/> )` and move down to just above the expectation on line 14.
You'll need to change `wrapper` to `cause `.

FedericoEsparza commented on src/components/Causes/Causes.test.js:
```diff
@@ -1,9 +1,14 @@
 import Causes from "./CausesContainer";
+import {shallow} from "enzyme"
 
+let wrapped
```
Please remove let

FedericoEsparza commented on src/components/Performances/Performances.test.js:
```diff
@@ -1,10 +1,16 @@
+
+  it("contains the text PerformancesContainer", ()=> {
+    expect(wrapped.text()).toEqual("PerformancesContainer")
+  })
```
Please remove beforeEach.

Modify to `const performances = shallow( < Performances/> )` and move down to just above the expectation on line 16.
You'll need to change `wrapper` to `performances `.

FedericoEsparza commented on src/components/Performances/Performances.test.js:
```diff
@@ -1,10 +1,16 @@
+import {shallow} from "enzyme"
 
-//Need at least 1 test to pass Travis CI
+let wrapped;
```
Please remove let

FedericoEsparza commented on src/components/Performance/Performance.test.js:
```diff
@@ -1,10 +1,16 @@
+
+  it("contains the text PerformanceContainer", ()=> {
+    expect(wrapped.text()).toEqual("PerformanceContainer")
+  })
```
Please remove beforeEach.

Modify to `const performance = shallow( < Performance/> )` and move down to just above the expectation on line 16.
You'll need to change `wrapper` to `performance `.

FedericoEsparza commented on src/components/Performance/Performance.test.js:
```diff
@@ -1,10 +1,16 @@
 import Performance from "./PerformanceContainer";
+import {shallow} from "enzyme"
+
+let wrapped;
```
Please remove let

FedericoEsparza commented on src/components/Home/Home.test.js:
```diff
@@ -1,9 +1,16 @@
+
+  it("contains the text HomeContainer", ()=> {
+    expect(wrapped.text()).toEqual("HomeContainer")
+  })
```
Please remove beforeEach.

Modify to `const home = shallow( < Home/> )` and move down to just above the expectation on line 16.
You'll need to change `wrapper` to `home `.

FedericoEsparza commented on src/components/Home/Home.test.js:
```diff
@@ -1,9 +1,16 @@
 import Home from "./HomeContainer";
+import {shallow} from "enzyme"
+
+let wrapped;
```
Please remove let

FedericoEsparza commented on src/components/Header/Header.test.js:
```diff
@@ -1,9 +1,16 @@
+
+  it("contains all the navigation links", ()=> {
+    expect(wrapped.find("NavLink").length).toEqual(6)
+  })
```
Please remove beforeEach.

Modify to `const header = shallow( < Header/> )` and move down to just above the expectation on line 16.
You'll need to change `wrapper` to `header `.

FedericoEsparza commented on src/components/Header/Header.test.js:
```diff
@@ -1,9 +1,16 @@
 import Header from "./Header";
+import {shallow} from "enzyme"
+
+let wrapped;
```
Please remove let

FedericoEsparza commented on src/components/Contact/Contact.test.js:
```diff
@@ -1,9 +1,16 @@
+  it('has text ContactContainer', ()=>{
+    expect(wrapped.text()).toEqual('ContactContainer')
+  })
+
```
Please remove beforeEach.

Modify to `const contact = shallow( < Contact/> )` and move down to just above the expectation on line 15.
You'll need to change `wrapper` to `contact `.

FedericoEsparza commented on src/components/Contact/Contact.test.js:
```diff
@@ -1,9 +1,16 @@
 import Contact from "./ContactContainer";
+import {shallow} from 'enzyme'
+
+let wrapped; 
```
Please remove let

FedericoEsparza commented on src/components/Causes/Causes.test.js:
```diff
@@ -1,9 +1,14 @@
+  it('contains text CausesContainer', ()=>{
+    expect(wrapped.text()).toEqual('CausesContainer')
+  })
+});
```
Please remove beforeEach.

Modify to `const causes = shallow( < Causes/> )` and move down to just above the expectation on line 14.
You'll need to change `wrapper` to `causes `.

FedericoEsparza reviewed (changes requested):
@aonomike @javpet @dhemmingson Great work on this.

Please make all requested modifications.

FedericoEsparza commented on src/components/Artist/Artist.test.js:
```diff
@@ -6,20 +6,20 @@ import {
-    expect(wrapped.find(MoneyRaised).length).toEqual(1)
+    expect(wrapper.find(MoneyRaised).length).toEqual(1)
   })
 });
```
Please modify `wrapped` to `artist`

FedericoEsparza commented on src/components/About/About.test.js:
```diff
@@ -1,9 +1,19 @@
+  shallow
+} from "enzyme";
+
+let wrapper;
```
Please remove let

FedericoEsparza commented on src/components/Artist/ArtistPerformanceList/ArtistPerfomanceList.test.js:
```diff
@@ -1,8 +1,18 @@
-    it('creates a component without crushing',() => {
-        const artistPerfomanceList = <ArtistPerfomenceList/>
-    })
+let wrapper
```
Please remove let

FedericoEsparza commented on src/components/Performances/Performances.test.js:
```diff
@@ -1,10 +1,18 @@
+} from "enzyme"
 
-//Need at least 1 test to pass Travis CI
+let wrapper;
```
Please remove let

FedericoEsparza commented on src/components/Performance/Performance.test.js:
```diff
@@ -1,10 +1,18 @@
+  shallow
+} from "enzyme"
+
+let wrapper;
```
Please remove let

FedericoEsparza commented on src/components/Home/Home.test.js:
```diff
@@ -1,9 +1,18 @@
+  shallow
+} from "enzyme"
+
+let wrapper;
```
Please remove let

FedericoEsparza commented on src/components/Header/Header.test.js:
```diff
@@ -1,9 +1,18 @@
+  shallow
+} from "enzyme"
+
+let wrapper;
```
Please remove let

FedericoEsparza commented on src/components/Contact/Contact.test.js:
```diff
@@ -1,9 +1,18 @@
-    const contact = <Contact/>;
-  });
-});
+let wrapper;
```
Please remove let

FedericoEsparza commented on src/components/Causes/Causes.test.js:
```diff
@@ -1,9 +1,16 @@
+  shallow
+} from "enzyme"
 
+let wrapper
```
Please remove let

FedericoEsparza commented on src/components/Cause/Cause.test.js:
```diff
@@ -1,9 +1,16 @@
+  shallow
+} from 'enzyme'
 
+let wrapper
```
Please remove let

FedericoEsparza commented on src/components/Artist/SupportedCauses/SupportedCauses.test.js:
```diff
@@ -1,8 +1,15 @@
-  it ("creates a component without crashing", () => {
-      const supportedCause = <SupportedCauses/>;
-  });
+let wrapper
```
Please remove let

FedericoEsparza commented on src/components/Artists/Artists.test.js:
```diff
@@ -1,9 +1,18 @@
+
+  it("contains the text 'ArtistsContainer'", () => {
+    expect(wrapper.find('div').text()).toEqual('ArtistsContainer');
   });
```
Please remove beforeEach.

Modify to `const artists = shallow( < Artists/> )` and move down to just above the expectation on line 16.
You'll need to change `wrapper` to `artists `.

FedericoEsparza commented on src/components/Artists/Artists.test.js:
```diff
@@ -1,9 +1,18 @@
+  shallow
+} from "enzyme"
+
+let wrapper;
```
Please remove let

FedericoEsparza commented on src/components/Artist/MoneyRaised/MoneyRaised.test.js:
```diff
@@ -1,9 +1,23 @@
+  it("contains the Chart component", () => {
+    expect(wrapper.find("Chart").length).toEqual(1)
+  })
+});
```
You'll need to change `wrapper` to `moneyRaised `.

FedericoEsparza commented on src/components/Artist/ArtistPerformance/ArtistPerformance.test.js:
```diff
@@ -1,10 +1,28 @@
 
-//Need at least 1 test to pass Travis CI
-
+let wrapper
```
Please move `let` to just above `beforeEach` and rename to `artistPerformance `

FedericoEsparza commented on src/components/Artist/MoneyRaised/MoneyRaised.test.js:
```diff
@@ -1,9 +1,23 @@
+  shallow
+} from "enzyme"
+
+let wrapper;
```
Please move `let` to just above `beforeEach` and rename to `moneyRaised`

FedericoEsparza commented on src/components/Artist/Artist.test.js:
```diff
@@ -6,20 +6,20 @@ import {
 } from "enzyme"
 
-let wrapped;
+let wrapper;
```
Please move `let` to just above `beforeEach` and rename to `artist `

FedericoEsparza commented on src/components/Artist/ArtistPerformanceList/ArtistPerfomanceList.test.js:
```diff
@@ -1,8 +1,18 @@
-describe('ArtistPerfomanceList', ()=>{
-    it('creates a component without crushing',() => {
-        const artistPerfomanceList = <ArtistPerfomenceList/>
-    })
```
Please re-add this test

FedericoEsparza commented on src/components/Cause/Cause.test.js:
```diff
@@ -1,9 +1,16 @@
-describe ("<CauseContainer />", () => {
-  it ("creates component without crashing", () => {
-    const cause = <Cause/>;
-  });
```
Please re-add this test

FedericoEsparza commented on src/components/Causes/Causes.test.js:
```diff
@@ -1,9 +1,16 @@
-describe ("<CausesContainer />", () => {
-  it ("creates component without crashing", () => {
-    const causes = <Causes/>;
-  });
```
Please re-add this test

FedericoEsparza commented on src/components/Contact/Contact.test.js:
```diff
@@ -1,9 +1,18 @@
-describe("<ContactContainer/>", ()=>{
-  it ("creates component without crashing", () => {
-    const contact = <Contact/>;
-  });
```
Please re-add this test

FedericoEsparza commented on src/components/Header/Header.test.js:
```diff
@@ -1,9 +1,18 @@
-describe("<Header/>", () => {
-  it ("creates component without crashing", () => {
-    const header = <Header/>;
-  });
```
Please re-add this test

FedericoEsparza commented on src/components/Home/Home.test.js:
```diff
@@ -1,9 +1,18 @@
-describe("<HomeContainer/>", () => {
-  it ("creates component without crashing", () => {
-    const home = <Home/>;
-  });
```
Please re-add this test

FedericoEsparza commented on src/components/Performance/Performance.test.js:
```diff
@@ -1,10 +1,18 @@
-"<PerfomanceContainer/>", () => {
-  it ("creates component without crashing", () => {
-    const performance = <Performance/>;
-  });
```
Please re-add this test

FedericoEsparza commented on src/components/Performances/Performances.test.js:
```diff
@@ -1,10 +1,18 @@
-  it ("creates component without crashing", () => {
-    const performances = <Performances/>;
-  });
-});
```
Please re-add this test

FedericoEsparza commented on src/components/About/About.test.js:
```diff
@@ -1,9 +1,19 @@
+
+  beforeEach(() => {
+    wrapper = shallow( < About / > )
+  })
```
Please remove beforeEach.

Please modify to `const about = shallow( < About/> )` and move down to just above the expectation on line 17.
You'll need to change `wrapper` to `about`.

FedericoEsparza reviewed (changes requested):
@aonomike @dhemmingson @javpet Review updated.

aonomike committed:
Refactor based on review

aonomike committed:
Update tests based on review

FedericoEsparza committed:
Merge branch 'develop' into 98_refactor-tests-to-use-enzyme

FedericoEsparza committed:
Re-insert beforeEach

Re-insert beforeEach, because we have more than one test, on this one, using `artistWrapper`

FedericoEsparza committed:
Re-insert beforeEach

FedericoEsparza committed:
Re-insert beforeEach

FedericoEsparza committed:
Rename to aboutWrapper

FedericoEsparza reviewed (approved):
@aonomike @javpet @dhemmingson Made a few modifications re-adding the beforeEach whenever you actually did have multiple tests using the wrapper.

Not the greatest fan of the use of the word `Wrapper`

I would prefer `componentName` instead, and as requested, but I guess I did say I was ok with `componentNameWrapper`, so I'll go ahead and approve.

**Thanks for your patience on the reviews, and for taking the time to make the modifications.**

I really appreciate it! 🥇 👍
